From a756f5266a97e4cdf0550283ec50210de58fbde3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Matt=C3=A9o=20Rossillol=E2=80=91=E2=80=91Laruelle?=
 <beatussum@protonmail.com>
Date: Mon, 17 Jun 2024 13:39:13 +0200
Subject: [PATCH] fix `tsc` call
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

In order to allow the meson build directory to be in another repertory (the
default behavior of the `meson` eclass), `--project` has to be added to the
`tsc` call.

Signed-off-by: Mattéo Rossillol‑‑Laruelle <beatussum@protonmail.com>
---
 src/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/meson.build b/src/meson.build
index e25ab8c..a8bbfb3 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -6,7 +6,7 @@ typescript = custom_target(
   input: files( 'main.ts' ),
   build_by_default: true,
   build_always_stale: true,
-  command: [ tsc, '--outDir', tsc_out ],
+  command: [ tsc, '--outDir', tsc_out, '--project', meson.source_root() ],
   output: ['tsc-output'],
 )
 
-- 
2.44.2

